### ISSUE TITLE
manage /tmp/raylogs usage: log rotation and permissions

### ISSUE THREAD

**Issue by atumanov** [ray-project/ray#526]
When using Ray on a publicly shared cluster, Ray creates /tmp/raylogs, which is currently not capped. Particularly in the case when something goes wrong, I've seen cases where we can easily run out of space in tmpfs. It would be good to implement log rotation to cap the amount of storage used by Ray logs.

A related issue is that the first user to create /tmp/raylogs may prevent others from running Ray on the same system, due to the 644 permissions (read-only for non-user) on /tmp/raylogs.

**Comment by atumanov** [comment]
```
---------------------------------------------------------------------------
IOError                                   Traceback (most recent call last)
<ipython-input-2-daa1e6a285d0> in <module>()
----> 1 ray.init()

/work/atumanov/ray/python/ray/worker.py in init(redis_address, node_ip_address, object_id_seed, num_workers, driver_mode, redirect_output, num_cpus, num_gpus)
   1060                num_workers=num_workers, driver_mode=driver_mode,
   1061                redirect_output=redirect_output, num_cpus=num_cpus,
-> 1062                num_gpus=num_gpus)
   1063 
   1064 

/work/atumanov/ray/python/ray/worker.py in _init(address_info, start_ray_local, object_id_seed, num_workers, num_local_schedulers, driver_mode, redirect_output, start_workers_from_local_scheduler, num_cpus, num_gpus)
    978         start_workers_from_local_scheduler=start_workers_from_local_scheduler,
    979         num_cpus=num_cpus,
--> 980         num_gpus=num_gpus)
    981   else:
    982     if redis_address is None:

/work/atumanov/ray/python/ray/services.py in start_ray_head(address_info, node_ip_address, num_workers, num_local_schedulers, worker_path, cleanup, redirect_output, start_workers_from_local_scheduler, num_cpus, num_gpus)
   1058       start_workers_from_local_scheduler=start_workers_from_local_scheduler,
   1059       num_cpus=num_cpus,
-> 1060       num_gpus=num_gpus)
   1061 
   1062 

/work/atumanov/ray/python/ray/services.py in start_ray_processes(address_info, node_ip_address, num_workers, num_local_schedulers, worker_path, cleanup, redirect_output, include_global_scheduler, include_redis, include_log_monitor, include_webui, start_workers_from_local_scheduler, num_cpus, num_gpus)
    823   if include_log_monitor:
    824     log_monitor_stdout_file, log_monitor_stderr_file = new_log_files(
--> 825         "log_monitor", redirect_output=True)
    826     start_log_monitor(redis_address,
    827                       node_ip_address,

/work/atumanov/ray/python/ray/services.py in new_log_files(name, redirect_output)
   1085   log_stdout = "{}/{}-{:06d}.out".format(logs_dir, name, log_id)
   1086   log_stderr = "{}/{}-{:06d}.err".format(logs_dir, name, log_id)
-> 1087   log_stdout_file = open(log_stdout, "a")
   1088   log_stderr_file = open(log_stderr, "a")
   1089   return log_stdout_file, log_stderr_file

IOError: [Errno 13] Permission denied: '/tmp/raylogs/log_monitor-052256.out'
```
One possibility is to set permissions on the created `raylogs` directory to be writeable by all: `os.chmod(path, 0777)`

**Comment by robertnishihara** [comment]
Addressed by #532.
